DL_NegotiationResponse.cpp

```diff
@@ -0,0 +1,143 @@
+    }
+
+    virtual std::string processMessage() const = 0;
+
+    // Set the flags (urgency, alert, response) based on chosen parameters
+    void setFlags(const std::string& urg, const std::string& alrt, const std::string& resp) {
+        urgency = urg;
+        alert = alrt;
+        response = resp;
+    }
+
+private:
+    // Convert intent enum to string
+    std::string getIntentName() const {
+        switch (intent) {
+            case MessageIntentUse:: WeCanAccept: return " WeCanAccept";
+            case MessageIntentUse::  WeCanNotAccept: return "  WeCanNotAccept";
+            default: return "UNKNOWN";
+        }
+    }
+
+    // Convert element enum to string
+    std::string getElementName() const {
+        switch (element) {
+            case MessageElement::URG: return "URG";
+            case MessageElement::ALRT: return "ALRT";
+            case MessageElement::RESP: return "RESP";
+            default: return "UNKNOWN";
+        }
+    }
+};
+
+// Downlink Message Class (Inherits from Message)
+class DownlinkMessage : public Message {
+public:
+    DownlinkMessage(MessageIntentUse intent, MessageElement element)
+        : Message(intent, element) {}
+
+    std::string processMessage() const override {
+        return "Processing  Request Voice Contact Frequency : " + getInfo();
+    }
+};
+
+// Function to choose Message Intent
+MessageIntentUse chooseIntent() {
+    int choice;
+    std::cout << "Choose Message Intent:\n"
+              << "1.  WeCanAccept\n"
+              << "2.   WeCanNotAccept\n";
+            
+    std::cin >> choice;
+
+    switch (choice) {
+        case 1: return MessageIntentUse:: WeCanAccept;
+        case 2: return MessageIntentUse::  WeCanNotAccept;
+        
+        default:
+            std::cout << "Invalid choice, defaulting to Request.\n";
+            return MessageIntentUse:: WeCanAccept;
+    }
+}
+
+// Function to choose Message Element
+MessageElement chooseElement() {
+    int choice;
+    std::cout << "Choose Message Element:\n"
+              << "1. URG\n"
+              << "2. ALRT\n"
+              << "3. RESP\n";
+    std::cin >> choice;
+
+    switch (choice) {
+        case 1: return MessageElement::URG;
+        case 2: return MessageElement::ALRT;
+        case 3: return MessageElement::RESP;
+        default:
+            std::cout << "Invalid choice, defaulting to URG.\n";
+            return MessageElement::URG;
+    }
+}
+
+// Function to set flags (N, M, L, etc.) for Downlink Message
+void setDownlinkFlags(DownlinkMessage& message) {
+    std::string urg, alrt, resp;
+    std::cout << "Enter URG flag (e.g., N, M): ";
+    std::cin >> urg;
+    std::cout << "Enter ALRT flag (e.g., L, M): ";
+    std::cin >> alrt;
+    std::cout << "Enter RESP flag (e.g., Y, N): ";
+    std::cin >> resp;
+
+    message.setFlags(urg, alrt, resp);
+}
+
+// Main function to run the program
+int main() {
```

Comment:
this main executions aren't needed , we just need the classes to be there

Reply:
I remove the main function